### ISSUE TITLE
Nomad plan always sees a change in the Job spec

### ISSUE THREAD

**Issue by apollo13** [hashicorp/nomad#8559]
### Nomad version
Nomad v0.12.1 (14a6893a250fc5f652eee4c7b52d8f95c3185aef)

### Issue
`nomad plan` always suggests changes for a job (namely NomadTokenID)

### Reproduction steps

Deploy the default example.nomad against your 0.12.1 cluster:
```
➜  nomad-jobs git:(master) ✗ nomad run example.nomad 
==> Monitoring evaluation "6d543761"
    Evaluation triggered by job "example"
    Evaluation within deployment: "74c52995"
    Allocation "ae41946f" created: node "8e100f4b", group "cache"
    Evaluation status changed: "pending" -> "complete"
==> Evaluation "6d543761" finished with status "complete"
➜  nomad-jobs git:(master) ✗ nomad plan example.nomad 
+/- Job: "example"
- NomadTokenID: "8d71945a-94e4-d236-9924-104810f09f56"
  Task Group: "cache" (1 in-place update)
    Task: "redis"

Scheduler dry-run:
- All tasks successfully allocated.

Job Modify Index: 127068
To submit the job with version verification run:

nomad job run -check-index 127068 example.nomad

When running the job with the check-index flag, the job will only be run if the
job modify index given matches the server-side version. If the index has
changed, another user has modified the job and the plan's results are
potentially invalid.
➜  nomad-jobs git:(master) ✗ nomad job run -check-index 127068 example.nomad
==> Monitoring evaluation "5c2078e4"
    Evaluation triggered by job "example"
    Evaluation within deployment: "74c52995"
    Evaluation status changed: "pending" -> "complete"
==> Evaluation "5c2078e4" finished with status "complete"
➜  nomad-jobs git:(master) ✗ nomad plan example.nomad                       
+/- Job: "example"
- NomadTokenID: "8d71945a-94e4-d236-9924-104810f09f56"
  Task Group: "cache" (1 in-place update)
    Task: "redis"

Scheduler dry-run:
- All tasks successfully allocated.

Job Modify Index: 127068
To submit the job with version verification run:

nomad job run -check-index 127068 example.nomad

When running the job with the check-index flag, the job will only be run if the
job modify index given matches the server-side version. If the index has
changed, another user has modified the job and the plan's results are
potentially invalid.
```

As you can see the output from plan always has `- NomadTokenID: "8d71945a-94e4-d236-9924-104810f09f56"`, and as such nomad thinks there are changes to make.

Might be a regression from https://github.com/hashicorp/nomad/pull/8204 /cc @tgross

**Comment by tgross** [comment]
Thanks for reporting this @apollo13, I'll take a look at it.

**Comment by tgross** [comment]
Hi @apollo13! I've just merged the fix in #8773 and this will ship in the upcoming 0.12.4.
